### PR TITLE
Vesting contract.

### PR DESCRIPTION
**Pull request by abancu**

### CODE REVIEW

File: contracts/Vesting.sol [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    function lastClaimedEpoch () public view returns (uint) {
+        return _lastClaimedEpoch;
+    }
+
+    function totalDistributedBalance () public view returns (uint) {
+        return _totalBalance;
+    }
+
+    function balance () public view returns (uint){
```
[the remark this answers is not in the document]
Comment (abancu) [reply contracts/Vesting.sol:53]: again, ma gandeam sa ii aratam userului cat mai are de incasat...

File: contracts/Vesting.sol [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        _bond = IERC20(bondTokenAddress);
+        _startTime = startTime;
+        totalDistributedBalance = totalBalance;
+    }
+
+    function claim () public onlyOwner {
+        uint balance;
+        uint currentEpoch = getCurrentEpoch();
+        if (currentEpoch > NUMBER_OF_EPOCHS + 1) {
+            currentEpoch = NUMBER_OF_EPOCHS + 1;
```
Comment (miladmostavi) [inline contracts/Vesting.sol:31]: I think here we should just transfer all the balance of the contract to owner (if > 0), in case there is any dust left.

**Comment by miladmostavi** [comment]
Can you please add a fallback function as well to do claims with 0 ETH txs to the contract?

```
fallback() external { claim(); }

**Comment by abancu** [comment]
> Can you please add a fallback function as well to do claims with 0 ETH txs to the contract?

done

**Review by miladmostavi — APPROVED** [review]

**Pull request merged**